lib/identity_cache.rb

```diff
@@ -72,12 +76,12 @@ def cache_backend=(cache_adaptor)
       if defined?(@cache)
         cache.cache_backend = cache_adaptor
       else
-        @cache = MemoizedCacheProxy.new(cache_adaptor)
+        @cache = wrap_in_replicated_proxy(MemoizedCacheProxy.new(cache_adaptor))
```

Comment:
Having MemoizedCacheProxy as the backend for the replicated proxy will mean that we will store duplicate keys in memory for all the replicas.  We want to reverse the wrapping so that we memoize the requested rather than replicated key to avoid extra memory use.

On the other hand, we don't want it as the cache backend, since we want a consistent interface to work with rather than having to worry about CacheFetcher and FallbackFetcher using this proxy differently.  As such, I think we should move the cache backend wrapping logic out of MemoizedCacheProxy so that we can add a layer between it and the CacheFetcher/FallbackFetcher.

It would also be nice to allow the user to control these proxies since that ability would allow what we are doing in this PR to be implemented outside of IdentityCache itself.  It would also allow us to add rollout code using a proxy layer that does double deleting while avoiding monkey patching.